### ISSUE TITLE
Does ATC work with tabular data?

### ISSUE THREAD

**Issue by blazespinnaker** [saurabhgarg1996/ATC_code#2]
I've been playing around with it a bit, and it seems to get to 100 accuracy pretty quickly and so doesn't provide a great signal.

fwiw, I've  found this code snippet below seems to provide better results (measured as correlation against ground truth scores) for ood tabular data that I've been looking at.  It also has an advantage of not requiring the model output for the holdout/validation data (just the unlabeled preds are needed), so that could be kept out (both input data and labels) if so required.

The code for non binary classification is a bit more complicated.

```
def get_ssl_metric(probas):
    #Now we're going to distill knowledge from the unlabeled test set
    model_eval = lightgbm.LGBMRegressor()
    #We use the SSL pseudolabels to train with and fit on unlabeled ood data
    model_eval.fit(X_test_unlabeled, probas)
    #Loop multiple times on unlabeled test data for enhanced knowledge density
    for j in range(5):
        print(".", end="")
        model_eval1 = lightgbm.LGBMRegressor()
        model_eval1.fit(X_test_unlabeled,model_eval.predict(X_test_unlabeled))
        model_eval = lightgbm.LGBMRegressor()
        model_eval.fit(X_test_unlabeled,model_eval1.predict(X_test_unlabeled))
    v = sklearn.metrics.roc_auc_score(y_holdout, model_eval.predict(X_holdout))
    return v
```

**Comment by saurabhgarg1996** [comment]
Thanks for your interest. Can you elaborate more about the tabular setup where you are observing 100% (predicted) accuracy? What is the true accuracy on the source and target dataset? What is the size of the labeled source holdout set and target unlabeled set? 

In our paper, we observe that our method ATC tends to perform well on natural distribution shift benchmarks. However, in general, the problem of estimating accuracy is underspecified (Section 3 of the paper https://arxiv.org/abs/2201.04234).

**Comment by blazespinnaker** [comment]
Would it make sense to evaluate this on non stationary time series tabular data?  eg:  https://www.kaggle.com/competitions/playground-series-s3e4/discussion/381415#2116636

The dataset is 314715 transactions, half for train.  The original dataset (non synthetic) is available for testing with ATC.
https://www.kaggle.com/datasets/mlg-ulb/creditcardfraud
